src/recursive_shared_mutex.h

```diff
@@ -0,0 +1,102 @@
+#ifndef _RECURSIVE_SHARED_MUTEX_H
+#define _RECURSIVE_SHARED_MUTEX_H
+
+#include <chrono>
+#include <tuple>
+#include <exception>
+#include <type_traits>
+#include <system_error>
+#include <thread>
+#include <condition_variable>
+#include <map>
+#include <mutex>
+#include <cassert>
+
+
+/*
+This mutex has two levels of access, shared and exclusive. Multiple threads can own this mutex in shared mode but only one can own it in exclusive mode.
```

Comment:
Preserve our Doxygen style commenting.

/**
 *  your text...
 *  your text..
 */

Reply:
fixed. i forgot to add the new file to the formatting list.